src/EnergyPlus/MixedAir.cc

```diff
@@ -1670,14 +1670,17 @@ namespace MixedAir {
 				for ( groupNum = 1; groupNum <= NumGroups; ++groupNum ) {
 					VentMechZoneName( groupNum ) = AlphArray( ( groupNum - 1 ) * 3 + 5 );
 
-					for ( OutAirNum = 1; OutAirNum <= NumOAControllers; ++OutAirNum ) {
-						if ( OAController( OutAirNum ).VentilationMechanicalName == VentilationMechanical( VentMechNum ).Name && VentilationMechanical( VentMechNum ).DCVFlag ) {
-							AirLoopControlInfo( OutAirNum ).AirLoopDCVFlag = true;
-						} else {
-							AirLoopControlInfo( OutAirNum ).AirLoopDCVFlag = false;
+					if ( AirLoopControlInfo.size() >= NumOAControllers ){
+						for ( OutAirNum = 1; OutAirNum <= NumOAControllers; ++OutAirNum ) {
+							if ( OAController( OutAirNum ).VentilationMechanicalName == VentilationMechanical( VentMechNum ).Name && VentilationMechanical( VentMechNum ).DCVFlag ) {
+								AirLoopControlInfo( OutAirNum ).AirLoopDCVFlag = true;
```

Comment:
Even without the alignment question, the logic is wrong here.  This block to set AirLoopDCVFlag should not be inside the groupNum for loop - it only needs to be done once per VentMech object.  And, the compound if should be split so that this takes action only when the name match is true.  As it stands now, all airloops but the last one will always have this flag set to false.  Posted a new issue #5489 to highlight the other problems here.  This pull request will address both issues with the same fixes.